src/Symfony/Component/Lock/Dsn.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock;
+
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+
+/**
+ * @author Konstantin Myakshin <molodchick@gmail.com>
+ * @author Hamza Amrouche <hamza.simperfit@gmail.com>
+ */
+final class Dsn
+{
+    private $scheme;
+    private $host;
+    private $user;
+    private $password;
+    private $port;
+    private $path;
+    private $options;
+
+    public function __construct(string $scheme, string $host, ?string $user = null, ?string $password = null, ?int $port = null, ?string $path = null, array $options = [])
+    {
+        $this->scheme = $scheme;
+        $this->host = $host;
+        $this->user = $user;
+        $this->password = $password;
+        $this->port = $port;
+        $this->path = $path;
+        $this->options = $options;
+    }
+
+    public static function isValid(string $dsn)
+    {
+        return false !== parse_url($dsn);
+    }
+
+    public static function fromString(string $dsn, array $options): self
+    {
+        if (false === $parsedDsn = parse_url($dsn)) {
+            throw new InvalidArgumentException(sprintf('The "%s" DSN is invalid.', $dsn));
+        }
+
+        parse_str($parsedDsn['query'] ?? '', $options);
+
+        return new self($parsedDsn['scheme'],
+            $parsedDsn['host'],
+            isset($parsedDsn['user']) ? urldecode($parsedDsn['user']) : null,
+            isset($parsedDsn['pass']) ? urldecode($parsedDsn['pass']) : null,
+            $parsedDsn['port'] ?? null, $parsedDsn['path'] ?? null,
+            $options);
+    }
+
+    public function getScheme(): string
```

Comment:
```suggestion
    public function scheme(): string
```

I am 👍 for no `get`-prefix. Same for the others